cmd/util/cmd/find-inconsistent-result/cmd.go

```diff
@@ -0,0 +1,164 @@
+package find_inconsistent_result
+
+import (
+	"fmt"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/spf13/cobra"
+
+	"github.com/onflow/flow-go/cmd/util/cmd/common"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+)
+
+var (
+	flagDatadir   string
+	flagEndHeight uint64
+)
+
+var Cmd = &cobra.Command{
+	Use:   "find-inconsistent-result",
+	Short: "find the first block that produces inconsistent results",
+	Run:   run,
+}
+
+func init() {
+	Cmd.Flags().StringVarP(&flagDatadir, "datadir", "d", "/var/flow/data/protocol", "directory to the badger dababase")
+	_ = Cmd.MarkPersistentFlagRequired("datadir")
+
+	Cmd.Flags().Uint64Var(&flagEndHeight, "end-height", 0, "the last block height checks for result consistency")
+	_ = Cmd.MarkPersistentFlagRequired("end-height")
+}
+
+func run(*cobra.Command, []string) {
+	height, err := findFirstMismatch(flagDatadir, flagEndHeight)
+	if err != nil {
+		fmt.Printf("fatal: %v\n", err)
+		return
+	}
+
+	fmt.Printf("first mismatch found at height %v\n", height)
+}
```

Comment:
I feel this implementation makes a very restrictive assumption when the tool is called:
* the engineer executing this tool knows there is a missmatch and tries to find out where it happened.
* But what if we use the tool for debugging, we are not sure if there is a mismatch and calling the tool to find out. We get the answer
   ```
   fatal: could not find first mismatch: no mismatch found between heights 74567 and 902360
  ```

I think the framing as `fatal` might send a very confusing message. Did the script complete and not find any errors? Or did it complete successfully and just determined that everything matches? If it is the latter, why the framing as fatal? 

I think this could be easily solved with a couple lines of extra code:
1. Introduce the sentinel 
   ```golang
   NoMissmatchFoundError := errors.New("No missmatch found")
   ```
2. In [line 112](https://github.com/onflow/flow-go/blob/8ca10bf4febdea0990e575adf9dc6b06fecdf8c8/cmd/util/cmd/find-inconsistent-result/cmd.go#L111-L112), we emit the sentinel 
   ```golang
   // No mismatch found within the range
   return 0, fmt.Errorf("no mismatch found between heights %v and %v: %w", startHeight, endHeight, err)
   ```
3. and here we do
```suggestion
	height, err := findFirstMismatch(flagDatadir, flagEndHeight)
	if err != nil {
		if errors.Is(err, NoMissmatchFoundError) {
			fmt.Printf(err.Error())
		}
		fmt.Printf("fatal: %v\n", err)
		return
	}
	fmt.Printf("first mismatch found at height %v\n", height)
}
```